code/controllers/subsystems/records.dm

```diff
@@ -0,0 +1,137 @@
+            r["notes"] = H.gen_record
+
+        //Medical Data
+        r["medical"]["blood_type"] = H.b_type
+        r["medical"]["blood_dna"] = H.dna.unique_enzymes
+        if(H.med_record && !jobban_isbanned(H, "Records"))
+            r["medical"]["notes"] = H.med_record
+
+        //Security Data
+        r["security"]["incidents"] = H.incidents
+        if(H.sec_record && !jobban_isbanned(H, "Records"))
+            r["security"]["notes"] = H.sec_record
+
+        //Locked Data
+        r["locked"] = r.Copy()
+        r["locked"]["nid"] = md5("[H.real_name][H.mind.assigned_role]")
+        r["locked"]["enzymes"] = H.dna.SE // Used in respawning
+        r["locked"]["identity"] = H.dna.UI
+        if(H.exploit_record && !jobban_isbanned(H, "Records"))
+            r["locked"]["exploit_record"] = H.exploit_record
+        else
+            r["locked"]["exploit_record"] = "No additional information acquired."
+
+        add_record(r)
+
+/datum/controller/subsystem/records/proc/add_record(var/list/record)
+    records[record["id"]] = record
+        
+/datum/controller/subsystem/records/proc/create_empty_record(var/mob/living/carbon/human/H, var/id)
+    var/icon/front
+    var/icon/side
+    if(H)
+        front = getFlatIcon(H, SOUTH, always_use_defdir = TRUE)
+        side = getFlatIcon(H, WEST, always_use_defdir = TRUE)
+    else
+        var/mob/living/carbon/human/dummy = new()
+        front = getFlatIcon(dummy, SOUTH, always_use_defdir = TRUE)
+        side = getFlatIcon(dummy, SOUTH, always_use_defdir = TRUE)
+        qdel(dummy)
+    
+    if(!id) id = generate_record_id()
+    var/list/r = list()
+    //General Data
+    r["name"] = "New Record"
+    r["id"] = id
+    r["rank"] = "Unassigned"
+    r["real_rank"] = "Unassigned"
+    r["sex"] = "Male"
+    r["age"] = "Unknown"
+    r["fingerprint"] = "Unknown"
+    r["phisical_status"] = "Active"
+    r["medical_status"] = "Stable"
+    r["species"] = "Human"
+    r["home_system"] = "Unknown"
+    r["citizenship"] = "Unknown"
+    r["faction"] = "Unknown"
+    r["religion"] = "Unknown"
+    r["photo_front"] = front
+    r["photo_side"] = side
+    r["notes"] = "No notes found."
+    r["ccia_record"] = "No CCIA records found"
+    r["ccia_actions"] = "No CCIA actions found"
+
+    //Medical Data
+    var/list/m = list()
+    r["medical"] = m
+    m["blood_type"] = "AB+"
+    m["blood_dna"] = md5("New Record")
+    m["disabilities"] = "No disabilities have been declared."
+    m["allergies"] = "No allergies have been detected in this patient."
+    m["diseases"] = "No diseases have been diagnosed at the moment."
+    m["notes"] = "No notes found."
+
+    //Security Data
+    var/list/s = list()
+    r["security"] = s
+    s["criminal"] = "None"
+    s["crimes"] = "There is no crime convictions."
+    s["notes"] = "No notes found."
+    s["incidents"] = ""
+
+    return r
+
+/datum/controller/subsystem/records/proc/find_record(var/field, var/value)
```

Comment:
This proc's usage is ambiguous and weird. For example, a call for `find_record("medical", some_char_name)` wouldn't work at all. If the intent is to return the full list with the record, why not simply neglect the field argument and simply accept a name?